pht_federated/aggregator/api/endpoints/discovery.py

```diff
@@ -0,0 +1,433 @@
+                "item_count": discovery_item_count,
+                "feature_count": discovery_feature_count,
+                "data_information": [discovery_summary_json_numeric]
+            }
+        elif feature_type == "unstructured":
+
+            discovery_summary_json_unstructured = {
+                "type": 'unstructured'
+            }
+            discovery_summary_schema = {
+                "proposal_id": proposal_id,
+                "item_count": discovery_item_count,
+                "feature_count": discovery_feature_count,
+                "data_information": [discovery_summary_json_unstructured]
+            }
+        elif feature_type == "unique":
+
+            discovery_summary_json_unique = {
+                "type": 'unique',
+                "title": feature_name,
+                "number_of_duplicates": discovery_no_duplicates
+            }
+            discovery_summary_schema = {
+                "proposal_id": proposal_id,
+                "item_count": discovery_item_count,
+                "feature_count": discovery_feature_count,
+                "data_information": [discovery_summary_json_unique]
+            }
+        elif feature_type == "equal":
+
+            discovery_summary_json_equal = {
+                "type": 'equal',
+                "title": feature_name,
+                "value": discovery_equal_value
+            }
+            discovery_summary_schema = {
+                "proposal_id": proposal_id,
+                "item_count": discovery_item_count,
+                "feature_count": discovery_feature_count,
+                "data_information": [discovery_summary_json_equal]
+            }
+
+        discovery_summary = DiscoverySummary(**discovery_summary_schema)
+
+        #print("DISCOVERY SUMMARY : {}".format(discovery_summary))
+
+        return discovery_summary
+
+@router.delete("/{proposal_id}/discovery", response_model=DiscoveryStatistics)
+def delete_discovery_statistics(proposal_id: int, db: Session = Depends(dependencies.get_db)) -> DiscoveryStatistics:
+    discovery_del = datasets.delete_by_dataset_id(proposal_id, db)
+    if not discovery_del:
+        raise HTTPException(status_code=404, detail=f"DatasetStatistics of proposal with id '{proposal_id}' not found.")
+    return discovery_del
+
+@router.post("/{proposal_id}/discovery", response_model=DiscoveryStatistics)
+def post_discovery_statistics(proposal_id: int, create_msg: StatisticsCreate, db: Session = Depends(dependencies.get_db)) -> DatasetStatistics:
+    dataset = json.loads(create_msg.json())
+    discovery_statistics_schema = {
+        "proposal_id": proposal_id,
+        "n_items": int(dataset['n_items']),
+        "n_features": int(dataset['n_features']),
+        "column_information": dataset['column_information']
+    }
+    discovery_statistics = DiscoveryStatistics(**discovery_statistics_schema)
+    discovery_statistics = datasets.create(db, obj_in=discovery_statistics)
```

Comment:
seems like unnecessary parsing, define input arguments and types directly/correctly.
why the int conversion?
dataset/s seems like the wrong name

datasets.create() should take StatisticsCreate as input